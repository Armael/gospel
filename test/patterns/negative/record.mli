type t = { n : int; s : string }

(*@ function f (x: t): unit =
    match x with
    | { n=p; s=("alpha"|"beta"|"gamma") } -> ()
    | { n=p; s="" } -> () *)
