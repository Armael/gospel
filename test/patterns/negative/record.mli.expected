
*******************************
********** Parsed file ********
*******************************
type t = {
  n: int ;
  s: string }
[@@@gospel
  {| function f (x: t): unit =
    match x with
    | { n=p; s=("alpha"|"beta"|"gamma") } -> ()
    | { n=p; s="" } -> () |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Stdlib *)

(*@ open Gospelstdlib *)

type t = {
n: int ;
s: string }
  

(*@ function f ... *)
File "record.mli", line 4, characters 4-90:
Error: This pattern-matching is not exhaustive.
       Here is an example of a case that is not matched:
         constr#t (0i, "?").
